private/bufpkg/bufcheck/bufbreaking/bufbreakingconfig/bufbreakingconfig.go

```diff
@@ -176,13 +176,18 @@ func configToJSON(config *Config) *configJSON {
 		})
 	}
 	sort.Slice(ignoreIDPathsJSON, func(i, j int) bool { return ignoreIDPathsJSON[i].ID < ignoreIDPathsJSON[j].ID })
-	sort.Strings(config.Use)
-	sort.Strings(config.Except)
-	sort.Strings(config.IgnoreRootPaths)
+	// We should not be sorting in place for the config structure, since it will mutate the
+	// underlying config ordering.
+	use := config.Use
```

Comment:
Unless I'm missing something, this doesn't change anything - slices are just pointers effectively, so this will continue to mutate the underlying slice

Reply:
Also, if we're doing this, it should be universal - we have the same problem with `rootPaths` above for example, if you want to say that `configToJSON` doesn't affect the underlying `Config` then we should be consistent about that (and add a comment to this effect)

Reply:
Ah, blergh, you're right -- I should be using `copy` here. Fixing this up.